packages/workspace/src/common/workspace-protocol.ts

```diff
@@ -12,12 +12,23 @@ export const workspacePath = '/services/workspace';
  */
 export const WorkspaceServer = Symbol('WorkspaceServer');
 export interface WorkspaceServer {
+
+    /**
+     * Returns with a promise that resolves to the workspace root URI as a string. Resolves to `undefined` if the workspace root is not yet set.
+     */
+    getRoot(): Promise<string | undefined>;
```

Comment:
as per comment below ... consider getRootURI

Reply:
not sure about it, for protocols URI is always `string`

Reply:
Yes the issue is more what is root ? is a URI or an fs path ? You have to open the code / jsdoc to figure it out.

Reply:
I don't think we send paths over JSON-RPC, only normalized URIs.